### DISCUSSION TITLE
How do you get this good?

### DISCUSSION THREAD

**Discussion by MarcusAseth** [z1dev/zkanji#42] [General]
> I'm not actively developing zkanji now, but I don't mind helping those who want to continue

I wish I could or had the skill, if I did, I know I have something that I would try and tweak/improve/add :D

But, of the things I learned as hobby about c++ in a year some years ago, now most of them are gone, and so I have no chance of achieving anything by putting hand on a program as monumental as zkanji and even if I remembered all those things that I knew, I still question if I was smart enough to deal with zkanji level of complexity eheh

But, just in case I can get some insight that might prove useful if I ever decide to seriously try again learning programming, I wanted to ask: how did you got this good? I mean I can't call myself a programmer at all but even I can see making zkanji is a feat that only someone that can be considered a Pro in this field could achieve
Also another question: if I where to seriously attempt this again (or when I'm ready for it), do you think there is something I could do or training/drill I should undergo in order the reach a level in which I could confidently attempt major modification to zkanji while having it still working? (not that I intend to anything like that, I'm just using it as a reference example for the skill level I would like to have)
Though probably I'm just looking for magic recepies/solution that do not exist and the only way is to spend an unbelievably high amount of hours doing it like on any other thing, be it learning japanese, how to draw or wathever :D

Still, I'm curious :)

**Comment by z1dev** [comment]
> Though probably I'm just looking for magic recepies/solution that do not exist and the only way is to spend an unbelievably high amount of hours doing it like on any other thing, be it learning japanese, how to draw or wathever :D

Yes, it's like learning Japanese!  Hey, if that's not inspiring then what is? :D The universal answer to this is that you have to start small with small goals, and then bigger ones with time. Trying to tackle something as big as zkanji right away will just burn you out. Although I don't think programming is inherently difficulty. It's just switching to a new way of thinking is difficult. If you mastered any language, doesn't have to be c++, if you did this right, seeing a new language for 10 seconds and you can code in it too.

I'm just repeating the same answer I have seen a million times from others too, and I don't know how helpful this is. Maybe it helps if you find a group of other beginners, or pros who are willing to help and make a project together, learning from each other as you go? Though that's only the way if you are an open and social person. You could try making simple games with a free engine, or reproduce a very small part of another program. Copying features is a great way to learn.

All in all, it's just time and effort and pain! :D

**Reply by MarcusAseth** [reply]
> All in all, it's just time and effort and pain! :D

what I had fearead xD
Thanks for your answer, I just have to wait for the right time then, I am terrible at multitasking so I wouldn't be able to mix well learning japanese with programming, but maybe in 3 to 5 years when I am N1 and hopefully above, I will be ready to start a new hobby again :)
